### PR TITLE
[Hexagon] Support both 1-d and 2-d VTCM allocations

### PR DESCRIPTION
**Pull request by Lunderberg**
This change allows allocation of buffers with `"global.vtcm"` to be either 1-d (flat) or 2-d (non-flat) memory allocations.

### CODE REVIEW

File: src/runtime/hexagon/hexagon/hexagon_buffer.h [hunk 1]
```diff
@@ -94,8 +94,24 @@ class HexagonBuffer {
...
-  //! \brief Return pointer to allocations.
-  void** GetPointer();
+  /*! \brief Return data pointer
+   *
+   * The return type depends on the buffer being
+   */
+  void* GetPointer();
+
+  // //! \brief Return number of allocations.
+  // size_t GetNumAllocs() { return nallocs_; }
```
Comment (adstraw) [inline src/runtime/hexagon/hexagon/hexagon_buffer.h:104]: remove

File: src/runtime/hexagon/hexagon/hexagon_buffer.h [hunk 2]
```diff
@@ -94,8 +94,24 @@ class HexagonBuffer {
...
+
+  /*! \brief Return dimensionality of buffer
+   *
+   * Will always be either 1 or 2.  If the buffer is 1-d, allocation
+   * returns a pointer to data, which is accessed by
+   * ((value_type*)ptr)[i].  If the buffer is 2-d, allocation returns
+   * a pointer to an array of pointers, which is accessed by
+   * ((value_type**)ptr)[i][j].
+   */
+  size_t GetBufferDimension() { return ndim_; }
```
Comment (adstraw) [inline src/runtime/hexagon/hexagon/hexagon_buffer.h:114]: remove, not called

**Review by adstraw — CHANGES_REQUESTED** [review]
LGTM; just need to clean up some unused code.

File: src/runtime/hexagon/hexagon/hexagon_buffer.h (hunk 2 above)
Comment (Lunderberg) [reply src/runtime/hexagon/hexagon/hexagon_buffer.h]: Thank you, and done.

File: src/runtime/hexagon/hexagon/hexagon_buffer.h (hunk 1 above)
Comment (Lunderberg) [reply src/runtime/hexagon/hexagon/hexagon_buffer.h]: Thank you, and done.

File: src/runtime/hexagon/hexagon/hexagon_buffer.h [hunk 3]
```diff
@@ -94,8 +94,11 @@ class HexagonBuffer {
   //! \brief Prevent move assignment.
   HexagonBuffer& operator=(HexagonBuffer&&) = delete;
 
-  //! \brief Return pointer to allocations.
-  void** GetPointer();
+  /*! \brief Return data pointer
+   *
+   * The return type depends on the buffer being
```
Comment (kparzysz-quic) [inline src/runtime/hexagon/hexagon/hexagon_buffer.h:99]: This comment looks truncated.

**Review by kparzysz-quic — APPROVED** [review]

**Pull request merged**